ndindex/ndindex.py

```diff
@@ -65,6 +66,91 @@ def __init__(self, f):
     def __get__(self, obj, owner):
         return self.f(owner)
 
+class _Guard:
+    def __init__(self):
+        self.val = False
+
+    def __call__(self):
+        if self.val:
+            return True
+        else:
+            self.val = True
+            return False
+
+def parse_index(node_or_string):
+    """
+    "Safely" (needs validation) evaluate an expression node or a string containing
+    a (limited subset) of valid numpy index or slice expressions.
+    """
+    if isinstance(node_or_string, str):
+        node_or_string = ast.parse('dummy[{}]'.format(node_or_string.lstrip(" \t")) , mode='eval')
+    if isinstance(node_or_string, ast.Expression):
+        node_or_string = node_or_string.body
+    if isinstance(node_or_string, ast.Subscript):
+        node_or_string = node_or_string.slice
+
+    def _raise_malformed_node(node):
+        raise ValueError(f'malformed node or string: {node!r}, {ast.dump(node)!r}')
```

Comment:
The `node` part here isn't particularly helpful alongside the dump.

Also I don't understand why the literal_eval error message says "node or string". How can the string be malformed? If there is a syntax error, it raises a different exception. I like this better:

```suggestion
        raise ValueError('unsupported node when parsing index: {ast.dump(node)}')
```

Reply:
I think your language is clearer in any case. I'll pull this suggestion in